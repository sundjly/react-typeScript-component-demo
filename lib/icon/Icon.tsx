import React from 'react';
import './importAll';
import './icon.scss';
import classnames from '../helpers/combineClass';

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> =
  ({className, name, ...ohersProp}) => {
    return (
      <svg
        className={classnames('customer-icon', className)}
        {...ohersProp}
      >
        <use xlinkHref={`#${name}`}/>
      </svg>
    );
  };

export default Icon;