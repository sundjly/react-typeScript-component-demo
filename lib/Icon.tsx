import React from 'react';
import './importAll';
import './icon.scss';

interface IconProps {
  name: string;
  onClick?: React.MouseEventHandler<SVGElement>
}

const Icon: React.FunctionComponent<IconProps> = (props) => {
  return (
    <svg className={'customer-icon'} onClick={props.onClick}>
      <use xlinkHref={`#${props.name}`}></use>
    </svg>
  );
};

export default Icon;