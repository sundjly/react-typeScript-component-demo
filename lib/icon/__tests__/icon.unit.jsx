import * as renderer from 'react-test-renderer';
import React from 'react';
import Icon from '../Icon';
import {mount} from 'enzyme'; // 测试点击事件引入的库

describe('icon组件测试', () => {
  it('icon render successfully', () => {
    const json = renderer.create(<Icon name="qq"/>).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('icon onClick', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="wechat" onClick={fn}/>);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});