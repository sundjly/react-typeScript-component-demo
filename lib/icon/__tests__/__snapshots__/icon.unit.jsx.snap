// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon组件测试 icon render successfully 1`] = `
<svg
  className="customer-icon"
>
  <use
    xlinkHref="#qq"
  />
</svg>
`;
